core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSink.java

```diff
@@ -79,4 +81,14 @@ default void closeIdleWriters() {}
     CompletableFuture<Collection<Slice>> finish();
 
     void abort();
+
+    /**
+     * Get the metrics for the page sink.
+     */
+    default Optional<PageSinkMetrics> getMetrics()
```

Comment:
it should be similar to `ConnectorPageSource`, see `io.trino.spi.connector.ConnectorPageSource#getMetrics`